tools/runners/run-test-suite.sh

```diff
@@ -154,7 +158,7 @@ do
 
         if [ $status_code -eq 0 ]
         then
-            test $VERBOSE && echo "[$tested/$TOTAL] $cmd_line: PASS"
+            test $VERBOSE && printf "[%4d/%4d] PASS: $cmd_line\n" "$tested" "$TOTAL"
```

Comment:
after this point, what is the use of `$cmd_line`? it is set at two following points, but as prints are rewritten to display the test name only, they are useless. however, the full command line is printed here, which makes the first assignment of cmd_line necessary.

note, this proposed approach makes the diagnostics somewhat inconsistent. the snapshot generation is printed in full command line, everything else just prints the js filename.

note 2, in case of a snapshot mode failure, it doesn't seem to be easy to tell whether generation or execution failed. the diagnostic message will be the same for both cases.

Reply:
It's a bit tricky because of the different cases of expected failures, for example if the snapshot generation should fail, or if the generation should pass but execution should fail, etc. How about something like this?
```sh
        if [ $status_code -eq 0 ]
        then
            test $VERBOSE && printf "[%4d/%4d] %bPASS: %s%b\n" "$tested" "$TOTAL" "$TERM_GREEN" "$SNAPSHOT_TEMP" "$TERM_NORMAL"

            $TIMEOUT_CMD $TIMEOUT $RUNTIME $ENGINE $ENGINE_ARGS --exec-snapshot $SNAPSHOT_TEMP &> $ENGINE_TEMP
            status_code=$?
        elif [ $error_code -eq 1 ]
        then
            test $VERBOSE && printf "[%4d/%4d] %bPASS (XFAIL): %s%b\n" "$tested" "$TOTAL" "$TERM_GREEN" "$SNAPSHOT_TEMP" "$TERM_NORMAL"
        else
            printf "[%4d/%4d] %bFAIL (%d): %s%b\n" "$tested" "$TOTAL" "$TERM_RED" "$status_code" "$SNAPSHOT_TEMP" "$TERM_NORMAL"
        fi
```
This way we will report both snapshot generation and execution (later in the other if block). `$cmd_line` is also deleted, so we will only display the name of the snapshot file, and now it's also colored.

Reply:
I don't think that we have had any concept for XFAIL of generation. And I don't think that this patch should rewrite the decision logic of the test runner. According to the PR title, this is about formatting improvements. So, I'd suggest no logic changes.

What I could think of is to have a `$comment` variable (or something like that), that's appended to `[index/total] outcome: test.js` outputs. For the various snapshot-related steps, it could contain ' (generate snapshot)' or ' (execute snapshot)', while it could be empty for non-snapshot executions.